### PR TITLE
[Modules] Enable AAD authentication

### PR DESCRIPTION
**Pull request by ppotturi**
# Description

Changes to enable AAD authentication and set AAD admins to the PostgreSQL flexibleServers.  Changed the  administratorLogin and administratorLoginPassword parameters to optional to set AAD authentication only

#3563 

### Pipeline references

| Pipeline |
| - |
| |

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/db-for-postgre-sql/flexible-server/administrator/main.bicep [hunk 1]
```diff
@@ -0,0 +1,65 @@
+metadata name = 'DBforPostgreSQL Flexible Server Administrators '
```
Comment (AlexanderSehr) [inline modules/db-for-postgre-sql/flexible-server/administrator/main.bicep:1]:
```suggestion
metadata name = 'DBforPostgreSQL Flexible Server Administrators'
```

**Comment by AlexanderSehr** [comment]
Hey @ppotturi , thank you very much for the added child-module.

Would you happen to have a test environment & pipeline badge available to show us that the change works as expected? This ideally means we should also add a test case where an instance with AAD auth is deployed.

If not, we could offer to open a dev branch on our end, merge the PR in there and implement the test on your behalf before merging to main.

File: modules/db-for-postgre-sql/flexible-server/administrator/main.bicep (hunk 1 above)
Comment (ppotturi) [reply modules/db-for-postgre-sql/flexible-server/administrator/main.bicep:1]: Fixed

**Comment by ppotturi** [comment]
> Hey @ppotturi , thank you very much for the added child-module.
> 
> Would you happen to have a test environment & pipeline badge available to show us that the change works as expected? This ideally means we should also add a test case where an instance with AAD auth is deployed.
> 
> If not, we could offer to open a dev branch on our end, merge the PR in there and implement the test on your behalf before merging to main.

I have no test environment or a pipeline to show. Please open a dev branch at your end and add tests please.

**Pull request merged**